metadatastore/commands.py

```diff
@@ -732,22 +1048,22 @@ def find_event_descriptors(run_start=None, **kwargs):
     -------
     event_descriptor : iterable of metadatastore.document.Document objects
     """
-    _format_time(kwargs)
-    _as_document = _AsDocument()
-    # get the actual mongo document
     if run_start:
-        run_start = _get_mongo_document(run_start, RunStart)
-        kwargs['run_start_id'] = run_start.id
+        run_start = doc_or_uid_to_uid(run_start)
+        run_start = runstart_given_uid(run_start)
+        run_start = _RUNSTART_UID_to_OID_MAP[run_start['uid']]
+        kwargs['run_start_id'] = run_start
+
+    _format_time(kwargs)
 
-    _normalize_object_id(kwargs, '_id')
-    _normalize_object_id(kwargs, 'run_start_id')
     event_descriptor_objects = EventDescriptor.objects(__raw__=kwargs)
 
-    event_descriptor_objects = event_descriptor_objects.no_dereference()
+    event_descriptor_objects = event_descriptor_objects.as_pymongo()
     for event_descriptor in event_descriptor_objects.order_by('-time'):
-        event_descriptor = _replace_descriptor_data_key_dots(event_descriptor,
-                                                             direction='out')
-        yield _as_document(event_descriptor)
+        yield _cache_descriptor(event_descriptor)
+
+# TODO properly mark this as deprecated
```

Comment:
If we are going to tag immediately after this is merged, I am very :-1: on waiting to mark this as deprecated